### PR TITLE
Feature/enh 44 add puppet tests

### PR DESCRIPTION
**Pull request by frodoslaw**
I added Puppet rspec cover tests. Please review.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/coi-gov-pl/puppet-jboss/pull/51" target="_blank">ReviewNinja

File: spec/classes/internal/configure/interfaces_spec.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  shared_examples 'completly working define' do
+    it { is_expected.to compile }
+    it { is_expected.to contain_user 'jboss' }
+    it { is_expected.to contain_class 'jboss' }
+    it { is_expected.to contain_group 'jboss' }
+    it { is_expected.to contain_class 'jboss::params' }
+    it { is_expected.to contain_class 'jboss::internal::runtime::dc' }
+    it { is_expected.to contain_class 'jboss::internal::configure::interfaces' }
+    it { is_expected.to contain_jboss__interface('public') }
+    it { is_expected.to contain_augeas('ensure present interface public') }
```
Comment (cardil) [inline spec/classes/internal/configure/interfaces_spec.rb]: !fix przynajmniej raz opisz parametry tych augeasów

File: spec/classes/internal/configure/interfaces_spec.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  shared_examples 'completly working define' do
+    it { is_expected.to compile }
+    it { is_expected.to contain_user 'jboss' }
+    it { is_expected.to contain_class 'jboss' }
+    it { is_expected.to contain_group 'jboss' }
+    it { is_expected.to contain_class 'jboss::params' }
+    it { is_expected.to contain_class 'jboss::internal::runtime::dc' }
+    it { is_expected.to contain_class 'jboss::internal::configure::interfaces' }
+    it { is_expected.to contain_jboss__interface('public') }
```
Comment (cardil) [inline spec/classes/internal/configure/interfaces_spec.rb]: !fix describe parameters of this `jboss::interface('public')` resource

File: spec/classes/internal/configure/interfaces_spec.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    it { is_expected.to contain_augeas('ensure present interface public') }
+    it { is_expected.to contain_augeas('interface public set any-address') }
```
Comment (cardil) [inline spec/classes/internal/configure/interfaces_spec.rb]: !fix describe parameters of this augeas resource at least once

File: spec/classes/internal/configure/interfaces_spec.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    it { is_expected.to contain_jboss__internal__interface__foreach("public:any-address") }
+
+    anchor_list.each do |item|
+      it { is_expected.to contain_anchor("#{item}") }
+    end
+    bind_variables_list.each do |var|
+      it { is_expected.to contain_augeas("interface public rm #{var}") }
```
Comment (cardil) [inline spec/classes/internal/configure/interfaces_spec.rb]: !fix describe parameters to at least one instance of this augeas

File: spec/defines/clientry_spec.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+describe 'jboss::clientry', :type => :define do
+  shared_examples 'completly working define' do
+    it { is_expected.to compile }
+    it { is_expected.to contain_jboss_confignode(title).with ({
+      :ensure => 'present',
+      :path   => 'profile/test'
+      }) }
+    it { is_expected.to contain_jboss_confignode(title).that_requires('Anchor[jboss::package::end]')}
+    it { is_expected.to contain_class 'jboss' }
+    it { is_expected.to contain_jboss__clientry(title) }
```
Comment (cardil) [inline spec/defines/clientry_spec.rb]: !fix describe parameters of this clientry, especially that you are testing this type!

File: spec/defines/deploy_spec.rb [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  shared_examples 'completly working define' do
+    it { is_expected.to compile }
+
+    it { is_expected.to contain_jboss_deploy(title).with ({
+      :ensure => 'present',
+      :source => '/tmp/jboss.war'
+      }) }
+    it {is_expected.to contain_class 'jboss' }
+    it {is_expected.to contain_class 'jboss::internal::runtime::node' }
+    it {is_expected.to contain_jboss__deploy(title) }
```
Comment (cardil) [inline spec/defines/deploy_spec.rb]: !fix describe parameters on `jboss::deploy` as you are testing this type

File: spec/defines/interface_spec.rb [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  shared_examples 'completly working define' do
+    it { is_expected.to compile }
+    it { is_expected.to contain_class 'jboss::params' }
+    it { is_expected.to contain_class 'jboss' }
+    it { is_expected.to contain_class 'jboss::internal::augeas' }
+    it { is_expected.to contain_class 'jboss::internal::runtime' }
+    it { is_expected.to contain_jboss__interface(title) }
+    it { is_expected.to contain_jboss__interface('public') }
+    bind_variables_list.each do |var|
+      it { is_expected.to contain_augeas("interface #{title} rm #{var}") }
```
Comment (cardil) [inline spec/defines/interface_spec.rb]: !fix describe augeas resources

**Comment by cardil** [comment]
@frodoslaw Please, describe parameters of all `augeas` resources as they are one of the most important ones

File: spec/classes/internal/configure/interfaces_spec.rb (hunk 1 above)
Comment (frodoslaw) [inline spec/classes/internal/configure/interfaces_spec.rb]: !fixed

File: spec/classes/internal/configure/interfaces_spec.rb (hunk 2 above)
Comment (frodoslaw) [inline spec/classes/internal/configure/interfaces_spec.rb]: !fixed

File: spec/classes/internal/configure/interfaces_spec.rb (hunk 3 above)
Comment (frodoslaw) [inline spec/classes/internal/configure/interfaces_spec.rb]: !fixed

File: spec/classes/internal/configure/interfaces_spec.rb (hunk 4 above)
Comment (frodoslaw) [inline spec/classes/internal/configure/interfaces_spec.rb]: !fixed

File: spec/defines/clientry_spec.rb (hunk 5 above)
Comment (frodoslaw) [inline spec/defines/clientry_spec.rb]: !fixed

File: spec/defines/deploy_spec.rb (hunk 6 above)
Comment (frodoslaw) [inline spec/defines/deploy_spec.rb]: !fixed

File: spec/defines/interface_spec.rb (hunk 7 above)
Comment (frodoslaw) [inline spec/defines/interface_spec.rb]: !fixed

**Comment by cardil** [comment]
Merge is completing #44

**Pull request merged**